packages/buckets/src/api/index.ts

```diff
@@ -664,12 +670,12 @@ export function bucketsPullPath(
   opts?: { progress?: (num?: number) => void },
   ctx?: ContextInterface,
 ): AsyncIterableIterator<Uint8Array> {
-  const metadata = { ...api.context.toJSON(), ...ctx?.toJSON() }
-  const request = new PullPathRequest()
-  request.setKey(key)
-  request.setPath(path)
-  let written = 0
-  const events = new EventIterator<Uint8Array>(({ push, stop, fail }) => {
+  return new Repeater<Uint8Array>((push, stop) => {
```

Comment:
Switching to the Repeater pattern here, which shouldn't affect any external APIs, but it doesn't simplify our code, and should be a more stable approach.